packages/protocol/contracts/L1/hooks/AssignmentHook.sol

```diff
@@ -120,7 +128,9 @@ contract AssignmentHook is EssentialContract, IHook {
                 refund = msg.value - input.tip;
             }
             // Paying ERC20 tokens
-            IERC20(assignment.feeToken).safeTransferFrom(msg.sender, blk.assignedProver, proverFee);
+            IERC20(assignment.feeToken).safeTransferFrom(
```

Comment:
This fixes the bug found by SigmaPrime:
> Also on the AssignmentHook when paying in fee tokens rather than native tokens it transfers from msg.sender which is actually the TaikoL1 contract. See here. This seems like it should be calling meta.coinbase or otherwise restricting it to the caller of blockPropose().